types/null/sf_point.go

```diff
@@ -0,0 +1,208 @@
+	p.Valid = false
+}
+
+// Interfaces
+
+// IsNil implements the pyrrho/encoding IsNiler interface. It will return true
+// if j is null.
+func (p SFPoint) IsNil() bool {
+	return !p.Valid
+}
+
+// IsZero implements the pyrrho/encoding IsZeroer interface. It will return true
+// if p is null or if the contained SFPoint is a zero value.
+func (p SFPoint) IsZero() bool {
+	if !p.Valid {
+		return true
+	}
+	return p.Point.IsZero()
+}
+
+// Value implements the database/sql/driver Valuer interface. It will return the
+// value of p as a driver.Value. If p is valid, this function will first
+// validate the contained SFPoint returning either any encouted parsing errors,
+// or a []byte as a driver.Value. If p is null, nil will be returned, and no
+// validation will occur.
+func (p SFPoint) Value() (driver.Value, error) {
+	if !p.Valid {
+		return nil, nil
+	}
+	return p.Point.Value()
+}
+
+// Scan implements the database/sql Scanner interface. It expects to receive a
+// valid WKB encoded byte describing a Point as a []byte, or NULL as a nil from
+// an SQL database. A zero-length []byte or a nil will be considered NULL, and p
+// will be nulled. Otherwise, the value will be passed to types.SFPoint to be
+// scanned and parsed as a WKB Point.
+func (p *SFPoint) Scan(src interface{}) error {
+	if p == nil {
+		return fmt.Errorf("null.SFPoint: Scan called on nil pointer")
+	}
+	switch x := src.(type) {
+	case nil:
+		p.Point = types.SFPoint{}
+		p.Valid = false
+		return nil
+	case []byte:
+		if len(x) == 0 {
+			p.Point = types.SFPoint{}
+			p.Valid = false
+			return nil
+		}
+		err := p.Point.Scan(x)
+		if err != nil {
+			p.Point = types.SFPoint{}
+			p.Valid = false
+			return err
+		}
+		p.Valid = true
+		return nil
+	default:
+		return fmt.Errorf("null.SFPoint: cannot scan type %T (%v)", src, src)
+	}
+}
+
+// MarshalJSON implements the encoding/json Marshaler interface. It will return
+// the value of p as a JSON-encoded []byte. If p is valid, this function will
+// first validate the contained JSON returning either any encouted parsing
+// errors, or a []byte. If p is null, "null" will be returned, and no validation
+// will occur.
+func (p SFPoint) MarshalJSON() ([]byte, error) {
+	if !p.Valid {
+		return []byte("null"), nil
+	}
+	return p.Point.MarshalJSON()
+}
+
+// UnmarshalJSON implements the encoding/json Unmarshaler interface. It expects
+// to receive a valid JSON value, and will assign that value to this SFPoint. If
+// the incoming JSON is the 'null' keyword, p will be nulled. UnmarshalJSON will
+// validate the incoming JSON as part of the "Is this JSON null?" check.
+func (p *SFPoint) UnmarshalJSON(data []byte) error {
+	if p == nil {
+		return fmt.Errorf("null.SFPoint: UnmarshalJSON called on nil pointer")
+	}
+	var k interface{}
+	if err := json.Unmarshal(data, &k); err != nil {
+		return err
+	}
+	if k == nil {
+		p.Point = types.SFPoint{}
+		p.Valid = false
+		return nil
+	}
+	p.Point.UnmarshalJSON(data)
```

Comment:
We should be capturing the returned error here, and only setting p.Valid = true if there wasn't a esploshun.